jira_integration/jira_integration.php

```diff
@@ -0,0 +1,111 @@
+<?php
+// TODO: this header should not be required...
+header('Content-Type: text/plain; charset=UTF-8');
+
+// Only watch the dev environment
+if ($_ENV['PANTHEON_ENVIRONMENT'] != 'dev') {
+  die();
+}
+
+// Retrieve jira destination and credentials
+$jira = json_decode(file_get_contents($_SERVER['HOME'] . '/files/private/jira_integration.json'), 1);
+if ($jira == FALSE) {
+  die('No jira file found. Aborting!');
+}
+
+$commits = array(
+  // The latest git hash
+  'current' => NULL,
+  // The hash at the last time we processed this script
+  'last_processed' => NULL,
+  // Raw output of git log since the last processed
+  'history_raw' => NULL,
+  // Formatted array of commits being sent to jira
+  'history' => array(),
+  // An array keyed by jira ticket id, each holding an
+  // array of commit ids.
+  'jira' => array()
+);
+
+// Get latest commit
+$commits['current'] = shell_exec('git rev-parse HEAD');
+
+// Retrieve the last commit processed by this script
+$commit_file = $_SERVER['HOME'] . '/files/private/jira_integration.commit';
+if (file_exists($commit_file)) {
+  $commits['last_processed'] = trim(file_get_contents($commit_file));
+}
+
+// Update the last commit file with the latest commit
+file_put_contents($commit_file, $commits['current'], LOCK_EX);
+
+// Retrieve git log for commits after last processed, to current
+$cmd = 'git log'; // add -p to include diff
+if (!$commits['last_processed']) {
+  $cmd .= ' -n 1';
+}
+else {
+  $cmd .= ' ' . $commits['last_processed'] . '...' . $commits['current'];
+}
+
+$commits['history_raw'] = shell_exec($cmd);
```

Comment:
We should fallback to simple `git log -n 1` if this fails because the commit range was invalid (e.g. after a force push). We should also echo out that this happened so it's in the log.